### DISCUSSION TITLE
How to dynamically load a local frontend project?

### DISCUSSION THREAD

**Discussion by kai230518** [tauri-apps/tauri#8285] [Q&A]
Hello, 
I'd like to know how I can dynamically load a local frontend project from the disk instead of compiling it into a binary file. I've tried using the file protocol, but it fails when the frontend communicates with Rust. I've also attempted to use RemoteDomainAccessScope, but communication still fails. Can someone guide me on how to achieve this, or is there a better approach?

**Comment by FabianLars** [comment]
I think the only thing that actually works is to spawn your own localhost server and serve the files from there. For simple use cases you may be able to use the `asset` protocol (basically the file protocol but for tauri apps) <https://tauri.app/v1/api/js/tauri/#convertfilesrc> but that won't get you _that_ far i think (never really tried it for this usecase).

Here's the tracking issue for this btw: https://github.com/tauri-apps/tauri/issues/5302

**Reply by kai230518** [reply]
Thank you, but I have another question. I want to use an iframe to load an HTML file (from an external plugin) that may have links to other JavaScript files (e.g., a Vue project). I saw in issue #7114 that iframes may not load external JavaScript correctly. Maybe the only solution is to load the web page from the external plugin into a local server, or is there a better approach?
